src/System.Runtime/tests/System/Double.cs

```diff
@@ -1,330 +1,402 @@
-    }
-
-    [Fact]
-    public static void TestToString()
+    [InlineData((double)789, (float)789, false)]
+    [InlineData((double)789, "789", false)]
+    public static void TestEquals(double d1, object value, bool expected)
     {
-        double i1 = 6310;
-        Assert.Equal("6310", i1.ToString());
-
-        double i2 = -8249;
-        Assert.Equal("-8249", i2.ToString());
+        if (value is double)
+        {
+            double d2 = (double)value;
+            Assert.Equal(expected, d1.Equals(d2));
+
+            if (double.IsNaN(d1) && double.IsNaN(d2))
+            {
+                Assert.Equal(!expected, d1 == d2);
+                Assert.Equal(expected, d1 != d2);
+            }
+            else
+            {
+                Assert.Equal(expected, d1 == d2);
+                Assert.Equal(!expected, d1 != d2);
+            }
+
+            Assert.NotEqual(0, d1.GetHashCode());
+            Assert.Equal(expected, d1.GetHashCode().Equals(d2.GetHashCode()));
+        }
+        Assert.Equal(expected, d1.Equals(value));
     }
 
-    [Fact]
-    public static void TestToStringFormatProvider()
+    public static IEnumerable<object[]> ToString_TestData()
     {
-        var numberFormat = new NumberFormatInfo();
-
-        double i1 = 6310;
-        Assert.Equal("6310", i1.ToString(numberFormat));
+        var emptyFormat = NumberFormatInfo.CurrentInfo;
+        yield return new object[] { double.MinValue, "G", emptyFormat, "-1.79769313486232E+308" };
+        yield return new object[] { (double)-4567, "G", emptyFormat, "-4567" };
+        yield return new object[] { -4567.89101, "G", emptyFormat, "-4567.89101" };
+        yield return new object[] { (double)0, "G", emptyFormat, "0" };
+        yield return new object[] { (double)4567, "G", emptyFormat, "4567" };
+        yield return new object[] { 4567.89101, "G", emptyFormat, "4567.89101" };
+        yield return new object[] { double.MaxValue, "G", emptyFormat, "1.79769313486232E+308" };
 
-        double i2 = -8249;
-        Assert.Equal("-8249", i2.ToString(numberFormat));
+        yield return new object[] { double.Epsilon, "G", emptyFormat, "4.94065645841247E-324" };
+        yield return new object[] { double.NaN, "G", emptyFormat, "NaN" };
 
-        double i3 = -2468;
+        yield return new object[] { (double)2468, "N", emptyFormat, string.Format("{0:N}", 2468.00) };
 
         // Changing the negative pattern doesn't do anything without also passing in a format string
-        numberFormat.NumberNegativePattern = 0;
-        Assert.Equal("-2468", i3.ToString(numberFormat));
-
-        Assert.Equal("NaN", double.NaN.ToString(NumberFormatInfo.InvariantInfo));
-        Assert.Equal("Infinity", double.PositiveInfinity.ToString(NumberFormatInfo.InvariantInfo));
-        Assert.Equal("-Infinity", double.NegativeInfinity.ToString(NumberFormatInfo.InvariantInfo));
+        var customFormat1 = new NumberFormatInfo();
+        customFormat1.NumberNegativePattern = 0;
```

Comment:
Consider using the:
```C#
var customFormat1 = new NumberFormatInfo { NumberNegativePattern = 0 };
```
syntax here and in other similar places, e.g. the one below could be:
```C#
var customFormat2 = new NumberFormatInfo {
    NegativeSign = "#",
    NumberDecimalSeparator = "~",
    NumberGroupSeparator = "*"
};
```